paho/client.go

```diff
@@ -145,21 +147,33 @@ func NewClient(conf ClientConfig) *Client {
 // returned. Otherwise the failure Connack (if there is one) is returned
 // along with an error indicating the reason for the failure to connect.
 func (c *Client) Connect(ctx context.Context, cp *Connect) (*Connack, error) {
+	if c.Conn == nil {
```

Comment:
I moved this `if` to the top because if we return right away then there's no need to initialize `cleanup`.